File: frontend/projects/admin/src/app/dto/zone-livraison.dto.ts
/**
 * Zone de livraison desservie.
 *
 * Les frais sont en FCFA (0 = livraison incluse) et le délai est exprimé
 * en heures, pas en texte libre : le backend contrôle la cohérence
 * delaiMin ≤ delaiMax.
 */
export interface ZoneLivraisonDTO {
  id?: number;
  nom: string;
  code: string;
  description?: string;
  fraisLivraison: number;
  delaiMinHeures: number;
  delaiMaxHeures: number;
  estActive: boolean;
  createdAt?: string;
  updatedAt?: string;
}

File: backend/src/main/java/com/vollailelink/backend/dto/ZoneLivraisonDTO.java
package com.vollailelink.backend.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.OffsetDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ZoneLivraisonDTO {
    private Long id;
    private String nom;
    private String code;
    private String description;
    /** Frais en FCFA. 0 = livraison incluse. */
    private BigDecimal fraisLivraison;
    private Integer delaiMinHeures;
    private Integer delaiMaxHeures;
    private Boolean estActive;
    private OffsetDateTime createdAt;
    private OffsetDateTime updatedAt;
}
